DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/board_test/src/CMakeLists.txt

```diff
@@ -22,6 +22,18 @@ if(NOT FPGA_DEVICE MATCHES ".*:.*")
     message(STATUS "This sample is made to target BSPs as this is a benchmarking sample.")
 endif()
 
+# Some tests in this sample require USM. Check the BSP name (which should contain the text 'usm')
+# and set a flag accordingly to disable those tests if the board does not support USM.
+# Allow the user to define USM_HOST_ALLOCATIONS_ENABLED to override this check
+# (e.g., cmake .. -DUSM_HOST_ALLOCATIONS_ENABLED=1)
+if((FPGA_DEVICE MATCHES ".usm.*") OR (DEFINED USM_HOST_ALLOCATIONS_ENABLED AND USM_HOST_ALLOCATIONS_ENABLED STREQUAL "1"))
+    set(USM_FLAG "-DSUPPORTS_USM")
+endif()
+
+if (USM_HOST_ALLOCATIONS_ENABLED)
+    message(STATUS "USM_HOST_ALLOCATIONS_ENABLED set manually!")
+endif()
+
 # This is a Windows-specific flag that enables error handling in host code
```

Comment:
we can detect if a device supports USM at runtime, i think it would be better to check at runtime rather than relying on checking the BSP filename :)

Reply:
Unfortunately, there is one part of the USM test that will not compile when targeting a non-USM board.  Specifically accessing a `sycl::host_ptr` pointer will fail with the following error:

```
usm_speed.hpp:25: Compiler Error: Could not resolve pointer to global memory. Please check that your chosen board target supports global memory or USM.
Error: Optimizer FAILED.
```

Reply:
We've decided that the best way to solve this is by forcing the user specify a compile-time macro "SUPPORTS_USM" in order to enable the USM tests. We will not try to enable them automatically by looking at the BSP name. Furthermore, we will perform a check at runtime to see if the board supports USM; if it does, and the user has not defined the SUPPORTS_USM macro, we will output a message suggesting they do so.